### DISCUSSION TITLE
auto_lighter.sc for ground level

### DISCUSSION THREAD

**Discussion by tanpohpoh** [gnembon/scarpet#273] [Q&A]
hihi, how would i go about editing the script so that it works for all solid blocks on surfaces above ground?

**Comment by TheDoctor1138** [comment]
Run the script like usual, (to light up caves)

then find this function in the script

if (__distance_sq(cpos, lpos) <= dd  
					&& air(lpos) && light(lpos) < 8 && sky_light(lpos) < 8
					&& solid(pos_offset(lpos, 'down')),
				if (is_survival && not_able_loose_torch(player),
					//running out of torches as survival
					return();
					
					
replace with this

if (__distance_sq(cpos, lpos) <= dd  
					&& air(lpos) && light(lpos) <= 15 && block_light(lpos) <= 0
					&& solid(pos_offset(lpos, 'down')),
				if (is_survival && not_able_loose_torch(player),
					//running out of torches as survival
					return();		
					

This is a crude way (that I found after several attempts)
it works kinda like this:

It checks if the average light level (block + sky) is equal or lower than 15, + if the block light (because for the surface, that's what matters) is 0

this way it doesn't spam the caves with torches and it removes any light level 0 spots (so for 1.18+ no mob spawning)		

Hope it helps, sorry for bad formatting
